### ISSUE TITLE
Is there any way to escape commas in comma-separated multiples->Seq?

### ISSUE THREAD

**Issue by javabrett** [scopt/scopt#105]
If using `unbounded` `opt` that accepts-multiple into a `Seq`, is there any way to express values that contain a comma `,`, i.e. escape it and prevent it from being interpreted as a value-delimiter, such that the option-value contains a comma?

**Comment by javabrett** [comment]
Could we adopt a convention whereby we split on `,`, except for escaping:
- `\,` does not split but becomes a literal `,` in a value so `a\,b,c` -> `("a,b","c")`
- `\\,` or `\\\\` or any other even number of backslashes becomes a literal backslash in the parsed value for each pair e.g. `\`, `\\`
- An odd number of backslashes preceding a comma therefore does not split and is some number of literal backslashes followed by a literal comma e.g. `a\\\\\,b` becomes `a\\,b` (single value)

**Comment by eed3si9n** [comment]
Anything in POSIX? http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html
Using conventional means of escape characters might become cumbersome because ones shell environment would also interpret them.

**Comment by javabrett** [comment]
Right, good point about shell-specials.  You'd end-up needing to quote these types of arguments to prevent the shell from treating them as escapes, which as you say could be cumbersome.

It is possible here that I've confused having an `opt[Seq[...` with the ability to allow `unbounded` of an option/type and collect the multiple values in a `Seq`.  The should be doable right?

At the moment I'm trying to use immutable parsing, and have a `someSeq: Seq[SomeClass] = Seq.empty` in my case class constructor.  In the `opt` `action` I  have:

```
opt[SomeClass]("my-option").abbr("O").text("description").action {
      (v, c) => c.copy(someSeq = c.someSeq:+v)
    }
```

... but for some reason I cannot fathom It is not working.

**Comment by javabrett** [comment]
I got immutable parsing of `unbounded` single-value options into a `Seq` working.  This now makes more-sense to me, to repeat the option rather than attempting to allow escaped-commas in the `opt[Seq]` variety.

I did crank out some code/tests for this, modifying the existing `sep` from a comma to something with a negative zero-width look-behind to check for backslashes.  It gets pretty complex around then allowing literal backslashes etc.  I'll stash that code but close this issue, since I think unbounded options is the best approach for this sort of problem.
